src/ucp/core/ucp_worker.c

```diff
@@ -454,14 +460,14 @@ static unsigned ucp_worker_iface_err_handle_progress(void *arg)
 
     ucs_debug("ep %p: handle error: %s", ucp_ep, ucs_status_string(status));
 
-    ucs_assert(ucp_ep->flags & UCP_EP_FLAG_FAILED);
+    ucs_assert(ucp_ep_flags & UCP_EP_FLAG_FAILED);
 
     ucp_ep_discard_lanes(ucp_ep, status);
     ucp_ep_reqs_purge(ucp_ep, status);
     ucp_stream_ep_cleanup(ucp_ep);
```

Comment:
user may close the EP from here

Reply:
maybe add the comment above the `ucp_stream_ep_cleanup()`?

if user closes EP, it will go to `if (ucp_ep_flags & UCP_EP_FLAG_USED)` and will invoke error callback for already closed EP